app/merginapi.cpp

```diff
@@ -1590,10 +1603,19 @@ void MerginApi::startProjectUpdate( const QString &projectFullName, const QByteA
     QString projectName;
     extractProjectName( projectFullName, projectNamespace, projectName );
 
+    // remove any leftover temp files that could be created from previous unsuccessful download
+    removeProjectsTempFolder( projectNamespace, projectName );
+
     // project has not been downloaded yet - we need to create a directory for it
     transaction.projectDir = createUniqueProjectDirectory( projectName );
     transaction.firstTimeDownload = true;
 
+    // create file indicating first time download in progress
+    QString downloadInProgressFilePath = InputUtils::downloadInProgressFilePath( transaction.projectDir );
+    createPathIfNotExists( downloadInProgressFilePath );
+    InputUtils::createFile( downloadInProgressFilePath );
+    Q_ASSERT( QFile::exists( downloadInProgressFilePath ) );
```

Comment:
I would rather log such thing instead of having assert (which will do nothing in release build and therefore we won't get any hint something went wrong)